src/pages/library/[buildingType]/index.jsx

```diff
@@ -1,11 +1,25 @@
 import { Breadcrumb, BreadcrumbItem, Flex, Text } from "@chakra-ui/react";
 import Link from "next/link";
 import { useRouter } from "next/router";
+import { useState } from "react";
+import { getCardsCount, getCardsPagination } from "server/mongodb/actions/Card";
 import CategoryCards from "src/components/CategoryCards";
+import SearchBar, { useSearch } from "src/components/SearchBar";
 import { buildingTypeNames } from "src/lib/utils/constants";
 
 function CategoriesPage({ buildingType }) {
   const router = useRouter();
+  const [cards, setCards] = useState([]);
+  const [setNumPages] = useState([]);
+  const [setCurrentPage] = useState(1);
+
+  const { handleSearch } = useSearch(
```

Comment:
idk why i can't comment on the on the `[categoryPage]` but make sure to pass in primaryCategory and buildingType in, cuz search is not working as intended there. i think its a false positive